swabbie-core-test/src/main/kotlin/com/netflix/spinnaker/handler/ResourceTypeHandlerTests.kt

```diff
@@ -0,0 +1,528 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.handler
+
+import com.netflix.spinnaker.config.Attribute
+import com.netflix.spinnaker.config.Exclusion
+import com.netflix.spinnaker.config.ExclusionType
+import com.netflix.spinnaker.config.NotificationConfiguration
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import com.netflix.spinnaker.swabbie.AbstractResourceTypeHandler
+import com.netflix.spinnaker.swabbie.events.DeleteResourceEvent
+import com.netflix.spinnaker.swabbie.events.MarkResourceEvent
+import com.netflix.spinnaker.swabbie.events.OwnerNotifiedEvent
+import com.netflix.spinnaker.swabbie.events.UnMarkResourceEvent
+import com.netflix.spinnaker.swabbie.exclusions.AllowListExclusionPolicy
+import com.netflix.spinnaker.swabbie.exclusions.LiteralExclusionPolicy
+import com.netflix.spinnaker.swabbie.model.Rule
+import com.netflix.spinnaker.swabbie.model.Status
+import com.netflix.spinnaker.swabbie.events.Action
+import com.netflix.spinnaker.swabbie.model.ResourceState
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import com.netflix.spinnaker.swabbie.model.MarkedResource
+import com.netflix.spinnaker.swabbie.model.Summary
+import com.netflix.spinnaker.swabbie.model.NotificationInfo
+import com.netflix.spinnaker.swabbie.repository.ResourceTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.TaskTrackingRepository
+import com.netflix.spinnaker.swabbie.repository.ResourceStateRepository
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.times
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.reset
+import com.nhaarman.mockito_kotlin.eq
+import com.nhaarman.mockito_kotlin.argWhere
+import com.nhaarman.mockito_kotlin.validateMockitoUsage
+import com.nhaarman.mockito_kotlin.doReturn
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.context.ApplicationEventPublisher
+import strikt.api.expectThat
+import strikt.assertions.isTrue
+import strikt.assertions.isFalse
+import strikt.assertions.isNull
+import strikt.assertions.isNotNull
+import strikt.assertions.isNotEmpty
+import strikt.assertions.isGreaterThan
+import strikt.assertions.isEqualTo
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneOffset
+
+abstract class ResourceTypeHandlerTests<T : AbstractResourceTypeHandler<*>> {
```

Comment:
Every handler test class must extend this base. Will convert the rest of the test in a separate PR